src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> header;
+    private Node<T> tail;
+    private int size;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            header = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is out of bounds");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, header);
+            header.prev = newNode;
+            header = newNode;
+        } else {
+            Node<T> currentPosition = spotNode(index);
+            Node<T> newNode = new Node<>(currentPosition.prev, value, currentPosition);
+            currentPosition.prev.next = newNode;
+            currentPosition.prev = newNode;
+        }
+        size++;
+    }
 
+    public Node<T> spotNode(int index) {
+        checkIndex(index);
+        Node<T> currentNode;
+        if (index <= size / 2) {
+            currentNode = header;
+            for (int i = 0; i != index; i++) {
+                currentNode = currentNode.next;
+            }
+        } else {
+            currentNode = tail;
+            for (int i = size - 1; i > index; i--) {
+                currentNode = currentNode.prev;
+            }
+        }
```

Comment:
Split in two different methods

Reply:
is there a sense to split? I would leave it as it is, the method is not large and not overloaded